e2e/test/scenarios/dashboard/dashboard.cy.spec.js

```diff
@@ -738,6 +743,42 @@ describe("scenarios > dashboard", () => {
   });
 });
 
+describeWithSnowplow.only("scenarios > dashboard", () => {
+  beforeEach(() => {
+    restore();
+    resetSnowplow();
+    cy.signInAsAdmin();
+    enableTracking();
+  });
+
+  afterEach(() => {
+    expectNoBadSnowplowEvents();
+  });
+
+  it("should track enabling the hide empty cards setting", () => {
+    visitDashboard(1);
+    editDashboard();
+
+    cy.findByTestId("dashboardcard-actions-panel").within(() => {
+      cy.icon("palette").click({ force: true });
+    });
+
+    cy.findByRole("dialog").within(() => {
+      cy.findByRole("switch", {
+        name: "Hide this card if there are no results",
+      })
+        .click() // enable
+        .click() // disable
+        .click(); // enable
+
+      // page_view
+      // card_set_to_hide_when_no_results first time
+      // card_set_to_hide_when_no_results second time
+      expectGoodSnowplowEvents(3);
```

Comment:
we added a snazzy new `expectGoodSnowplowEvent` helper where you can actually test event names and numbers

https://github.com/metabase/metabase/blob/58c7b0ffa56c53f153a1fc4ecc896f60fb899a4e/e2e/support/helpers/e2e-snowplow-helpers.js#L33-L41

sample usage:

https://github.com/metabase/metabase/blob/58c7b0ffa56c53f153a1fc4ecc896f60fb899a4e/e2e/test/scenarios/collections/uploads.cy.spec.js#L126-L128

I think here it will be

```ts
expectGoodSnowplowEvent({
  event: "card_set_to_hide_when_no_results"
}, 2);
```